### PR TITLE
adjusting mnemonics validation

### PR DESCRIPTION
**Pull request by maayarosama**
### Description

Mnemonics doesn't accept special characters.

### Related Issues

#1048

### CODE REVIEW

**Review by AhmedHanafy725 — COMMENTED** [review]
it accepts less than 25 words
image

File: src/utils/validateName.ts [hunk 1]
```diff
@@ -15,7 +15,8 @@ const URL_REGEX =
   /^((?:(?:http?|ftp)[s]*:\/\/)?[a-z0-9-%\/\&=?\.]+\.[a-z]{2,4}\/?([^\s<>\#%"\,\{\}\\|\\\^\[\]`]+)?)/;
 const WHITE_SPACE_REGEX = /^\S*$/;
 const NUM_REGEX = /^[1-9](\d?|\d+)$/;
-const MNEMONICS_LENGTH_REGEX= /^([a-zA-Z0-9]+\s?){25}$/;
+const MNEMONICS_LENGTH_REGEX= /^(?:\s*[a-zA-Z]+(?:\s+[a-zA-Z]+){24})?\s*$/;
```
Comment (xmonader) [inline src/utils/validateName.ts]: you can use `\w` meta for `a-zA-Z0-9A-Z`, but I assume you want to exclude 0-9, `^([a-zA-Z_]+\s?){25}$` this should work.

Comment (maayarosama) [reply src/utils/validateName.ts]: ^([a-zA-Z_]+\s?){25}$ this regex accepts less that 25 words, as it's counting whitespaces as words.

File: src/utils/validateName.ts [hunk 2]
```diff
@@ -38,12 +39,10 @@ export function validateMnemonics(mnemonics: string): string | void {
 
 export function validateMnemonicsAlgorand(mnemonics: string): string | void {
   if (!mnemonics.length) return "Mnemonics required";
-  if (!ALPHA_ONLY_REGEX.test(mnemonics))
-    return "Mnemonics are can only be composed of a non-alphanumeric character or a whitespace.";
   if (!MNEMONICS_LENGTH_REGEX.test(mnemonics))
-    return "Mnemonics should be 25 words";
-}
+    return "Mnemonics should be 25 words and shouldn't contain alphanumeric characters.";
```
Comment (AhmedHanafy725) [inline src/utils/validateName.ts]:
```suggestion
    return "Mnemonics should be 25 words and should contain alphabetic characters with only one space between the words.";
```

**Review by AhmedHanafy725 — APPROVED** [review]

**Pull request merged**